### PR TITLE
Sdk 2652 go implement sdk helper methods for digital id estimated age attribute

### PR DESCRIPTION
**Pull request by mehmet-yoti**
# SDK-2652: Implement SDK helper methods for Digital ID estimated age attribute

## Summary

This PR implements comprehensive helper methods for Relying Parties to easily request the `estimated_age` attribute with automatic fallback to the `date_of_birth` attribute from Digital ID app shares. The implementation enhances user experience by requesting the most specific age information available while maintaining robust age verification capabilities.

#### Policy Builder Extensions
- **Digital Identity Policy Builder (`digitalidentity` package)**:
  - `WithEstimatedAge(options...)` - Requests estimated_age with date_of_birth fallback
  - `WithEstimatedAgeOver(age int, options...)` - Age over verification with fallback
  - `WithEstimatedAgeUnder(age int, options...)` - Age under verification with fallback

- **Dynamic Policy Builder (`dynamic` package)**:
  - `WithEstimatedAge(options...)` - Requests estimated_age with date_of_birth fallback  
  - `WithEstimatedAgeOver(age int, options...)` - Age over verification with fallback
  - `WithEstimatedAgeUnder(age int, options...)` - Age under verification with fallback

#### User Profile Helper Methods
- **Digital Identity User Profile**: 
  - `EstimatedAge()` - Returns the estimated_age attribute if present
  - `EstimatedAgeWithFallback()` - Returns estimated_age or falls back to date_of_birth

- **Profile User Profile**:
  - `EstimatedAge()` - Returns the estimated_age attribute if present  
  - `EstimatedAgeWithFallback()` - Returns estimated_age or falls back to date_of_birth

#### Comprehensive Documentation
- **`ESTIMATED_AGE_HELPERS.md`** - Complete guide with usage examples, API reference, and implementation details
- **`_examples/digitalidentity/README.md`** - Updated with estimated age examples and setup instructions

#### Production-Ready Examples
- **`estimated_age_example.go`** - Complete Go backend implementation with all 4 verification scenarios
- **`estimated_age_examples.html`** - Interactive HTML frontend with Yoti Web SDK integration
- **`estimated_age_receipt.go`** - Enhanced receipt processing with age verification data extraction
- **`estimated_age_receipt.html`** - Comprehensive receipt display with JSON debugging capabilities

#### Core Constants & Attributes
- Added `AttrEstimatedAge` constant for consistent attribute naming
- Enhanced `WantedAttributeBuilder` with `AlternativeNames` support for fallback logic
- Support for variadic constraint parameters (source constraints, document type verification)

#### Robust Error Handling
- Comprehensive input validation for age parameters
- Graceful fallback mechanisms when estimated_age is unavailable
- Detailed error messages for debugging and troubleshooting

#### Comprehensive Test Suite
- **Unit Tests**: 137 test cases covering all helper methods and edge cases
- **Integration Tests**: Real-world scenarios with policy building and receipt processing  
- **Constraint Tests**: Document type verification and source constraint validation
- **Fallback Tests**: Automatic fallback behavior verification

#### Test Coverage Areas
- Policy builder method validation
- User profile attribute extraction
- Age verification logic (over/under scenarios)
- Constraint application and validation
- Error handling and edge cases

### 🎯 Automatic Fallback Logic
The implementation intelligently requests both `estimated_age` and `date_of_birth` attributes, prioritizing estimated_age while automatically falling back to date_of_birth when unavailable.

### 🛡️ Source Constraint Support
All methods support optional source constraints for enhanced security:
```go
policy, err := (&digitalidentity.PolicyBuilder{}).
    WithEstimatedAgeOver(18, passportConstraint).
    Build()

### CODE REVIEW

File: yotiSelfSignedKey.pem [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
[the remark this answers is not in the document]
Comment (Alttaf) [reply yotiSelfSignedKey.pem:1]: this looks like it should be in a serets file, is this intentional?
